codegenerator/cli/templates/static/codegen/src/eventFetching/ChainManager.res

```diff
@@ -92,22 +115,16 @@ let determineNextEvent = (
       // If the fetch state has reached the end block we don't need to consider it
       switch partitionedFetchState->PartitionedFetchState.getEarliestEvent {
       | Some(earliestEvent) =>
-        let {val, isInReorgThreshold} = accum
-        let mk = cmp => {
+        let current: multiChainEventComparitor = {chain, earliestEvent}
+        switch accum.val {
+        | Some(previous) if comparitorFunction(previous, current) => accum
+        | _ =>
+          let isInReorgThreshold =
+            earliestEvent->FetchState.queueItemIsInReorgThreshold(~heighestBlockBelowThreshold)
+
           {
-            val: Some(cmp),
-            isInReorgThreshold: isInReorgThreshold ||
-            cmp.earliestEvent->FetchState.queueItemIsInReorgThreshold(~heighestBlockBelowThreshold),
-          }
```

Comment:
This was buggy, it was accumulating "isInReorgThreshold" across all chains. Once the comparisons are done it should only return "isInReorgThreshold" with relation to the chain of the item that it's returning.